crates/services/txpool_v2/src/pending_pool.rs

```diff
@@ -0,0 +1,585 @@
+use std::{
+    collections::{
+        hash_map::Entry,
+        HashMap,
+        HashSet,
+        VecDeque,
+    },
+    time::{
+        Duration,
+        SystemTime,
+    },
+};
+
+use fuel_core_types::{
+    fuel_tx::{
+        ContractId,
+        Output,
+        TxId,
+        UtxoId,
+    },
+    services::txpool::ArcPoolTx,
+};
+
+use crate::{
+    error::{
+        Error,
+        InputValidationError,
+    },
+    pool_worker::InsertionSource,
+};
+
+// This is a simple temporary storage for transactions that doesn't have all of their input created yet.
+// This storage should not have a lot of complexity.
+//
+// Insertion rules:
+// - If the transaction has one or more inputs that are not known yet.
+//
+// Deletion rules:
+// - If the transaction missing inputs hasn't been known for a certain amount of time.
+// - If the transaction missing inputs becomes known.
+pub(crate) struct PendingPool {
+    ttl: Duration,
+    pending_txs_by_inputs: HashMap<MissingInput, HashSet<TxId>>,
+    pending_inputs_by_tx:
+        HashMap<TxId, ((ArcPoolTx, InsertionSource), Vec<MissingInput>)>,
```

Comment:
It would be nice to use some structure instead of `((ArcPoolTx, InsertionSource), Vec<MissingInput>)`, hard to remember where do we store what=D

Reply:
Also, why do we use`((ArcPoolTx, InsertionSource), Vec<MissingInput>)` instead of ` (ArcPoolTx, InsertionSource, Vec<MissingInput>)`?

Reply:
Done.

Leftover from previous version.